src/codeflare_sdk/ray/cluster/generate_yaml.py

```diff
@@ -88,9 +94,15 @@ def update_names(
 
 def update_image(spec, image):
     containers = spec.get("containers")
-    if image != "":
-        for container in containers:
-            container["image"] = image
+    if not image:
+        python_version = f"{sys.version_info.major}.{sys.version_info.minor}"
+        try:
+            if python_version in SUPPORTED_PYTHON_VERSIONS:
+                image = SUPPORTED_PYTHON_VERSIONS[python_version]
+        except Exception: # pragma: no cover
+            print(f"Python version '{python_version}' is not supported. Only {', '.join(SUPPORTED_PYTHON_VERSIONS.keys())} are supported.")
```

Comment:
Perhaps a warning in case the user knows what they are doing/has a custom image and still wants to proceed.
If we could be more concise and say "officially supported" that might help too.

Reply:
Also in the warning maybe a "If you would like to proceed please specify an image or use one of the supported vers"